www/roster/main.rb

```diff
@@ -167,6 +167,14 @@
   _html :podlings
 end
 
+# gives a list of podlings that would report in a given month based on current status
+get '/podlings/reporting/:month.json' do |month|
```

Comment:
The only real issue here is that this particular tool require authentication; that will make it less easy to use as input to another script.  The code below can be taken pretty much as is (OK, you will need to extract the month from the URL or params for yourself) and made a standalone CGI.